src/scripts/classes/Transitions.ts

```diff
@@ -105,9 +107,18 @@ export class Transitions {
      * @see https://swup.js.org/hooks/#visit-start
      * @param visit: VisitType
      */
-    onVisitStart() {
+    onVisitStart(visit: VisitType) {
         document.documentElement.classList.add(Transitions.TRANSITION_CLASS);
         document.documentElement.classList.remove(Transitions.READY_CLASS);
+
+        if (visit.a11y) {
+            visit.a11y.focus = 'h1#page-heading';
```

Comment:
So, Swup is the one at fault for the recent issue I noticed across some projects where the page does not scroll back to the top after clicking a hyperlink in the footer.

Swup created a problem and instead of fixing it introduces a non-standard solution.

> But it seems to prevent Swup's `aria-live` announcements to happen as a side effect... Not sure if that's a real issue though as the plugin announces `Navigated to {title}` by default […]

If its announcing `Navigated to {title}` that means its `aria-live` element is working as intended. It's extracting the text node from the first `<h1>` which happens to be the target of your `visit.a11y.focus`.

Since the `<h1>` can't be guaranteed to have a relevant title, maybe the plugin's `headingSelector` could target `<title>` instead.

Reply:
In regards to `visit.a11y.focus`, that makes sense that it's stealing focus. Not necessarily the intended behavior. I noticed that Swup's documentation site still uses v4 whose behaviour is to focus the first `h1`.